python/lsst/ci/build.py

```diff
@@ -344,20 +347,22 @@ def _build_product_if_needed(self, product):
             try:
                 # skip the build if the product has been installed
                 eups_prod, retcode, logfile = self.eups.getProduct(product.name, product.version), 0, None
+                fetched = False
             except eups.ProductNotFound:
                 distrib_path = os.environ["EUPS_DISTRIB"]
 
                 if not self.no_fetch_binary and self._check_if_in_distrib(product, distrib_path):
                     eups_prod, retcode, logfile = self._fetch_product_if_needed(
                         product, distrib_path, progress
                     )
+                    fetched = True
                 else:
                     eups_prod, retcode, logfile = self._build_product(product, progress)
 
             if eups_prod is not None and self.manifest.build_id not in eups_prod.tags:
                 self._tag_product(product.name, product.version, self.manifest.build_id)
 
-            progress.report_result(retcode, logfile)
+            progress.report_result(retcode, logfile, fetched)
```

Comment:
Pretty sure that there is a code path where `fetched` is undefined (the try fails and triggers the except and then the else branch is used). I think you need to declare it before the try block on line 347.